scripts/sas_ou_gpo_indicators.sh

```
@@ -0,0 +1,205 @@
+  local stage="$1" level="$2" msg="$3" data="${4:-}" ts; ts="$(now_iso)"
+  printf '{"timestamp":"%s","run_id":"%s","stage":"%s","level":"%s","message":"%s","data":"%s"}\n' \
+    "$(json_escape "$ts")" "$(json_escape "$RUN_ID")" "$(json_escape "$stage")" "$(json_escape "$level")" "$(json_escape "$msg")" "$(json_escape "$data")" >> "$EVENT_LOG"
+  printf '[%s][%s] %s\n' "$stage" "$level" "$msg" | tee -a "$TRACE_LOG"
+}
+
+capture() {
+  local label="$1" out="$2"; shift 2
+  { echo "# $label"; echo "# command: $*"; echo "# captured_at: $(now_iso)"; echo; } > "$out"
+  printf '%s\n' "$*" >> "$PLAN"
+  if [[ "$DRY_RUN" -eq 1 ]]; then echo "DRY RUN" >> "$out"; echo "# exit_code: 0" >> "$out"; return 0; fi
+  set +e; "$@" >> "$out" 2>&1; code=$?; set -e
```

Shellcheck (0.11.0)

[info] This is actually an end quote, but due to next char it looks suspect.

(SC1079)